src/core/ddsc/tests/psmxif.c

```diff
@@ -0,0 +1,441 @@
+      CU_ASSERT_FATAL(dmock->cnt_create_psmx == 1); // Confirm the dummy psmx has been loaded.
+      dds_entity_unpin(&dom->m_entity);
+    }
+    // Check that the config string passed to `dds_create_domain()` has been correctly forwarded to the dummy psmx.
+    CU_ASSERT_FATAL(strstr(configstr_in, dmock->config) != NULL); // dmock->config is a substring of the original xml.
+    ddsrt_free(configstr_in);
+
+    dds_entity_t writer1 = 0, reader1 = 0, writer2 = 0, reader2 = 0;
+    {
+      char topicname[100];
+      dummy_topics_alloc(dmock, 2);
+      dummy_endpoints_alloc(dmock, 4);
+
+      create_unique_topic_name("shared_memory", topicname, sizeof(topicname));
+      psmx_topic_expected = (dds_psmx_topic_t*)dmock->topics._buffer + dmock->topics._length;
+      dds_entity_t topic1 = dds_create_topic(participant, &SC_Model_desc, topicname, NULL, NULL);
+      CU_ASSERT_FATAL(topic1 > 0);
+      {
+        dds_entity* x = NULL;
+        dds_return_t rc = dds_entity_pin(topic1, &x);
+        CU_ASSERT_FATAL(rc == DDS_RETCODE_OK && dds_entity_kind(x) == DDS_KIND_TOPIC);
+        struct dds_psmx_topics_set* topics_set = &((dds_topic*)x)->m_ktopic->psmx_topics;
+        CU_ASSERT_FATAL(topics_set->length == 1 && topics_set->topics[0] == psmx_topic_expected);
+        dds_entity_unpin(x);
+      }
+
+      psmx_endpt_expected = (dds_psmx_endpoint_t*)dmock->endpoints._buffer + dmock->endpoints._length;
+      writer1 = dds_create_writer(participant, topic1, NULL, NULL);
+      CU_ASSERT_FATAL(writer1 > 0);
+      {
+        dds_entity* x = NULL;
+        dds_return_t rc = dds_entity_pin(writer1, &x);
+        CU_ASSERT_FATAL(rc == DDS_RETCODE_OK && dds_entity_kind(x) == DDS_KIND_WRITER);
+        struct dds_psmx_endpoints_set* endpt_set = &((dds_writer*)x)->m_endpoint.psmx_endpoints;
+        CU_ASSERT_FATAL(endpt_set->length == 1 && endpt_set->endpoints[0] == psmx_endpt_expected);
+        dds_entity_unpin(x);
+      }
+
+      psmx_endpt_expected = (dds_psmx_endpoint_t*)dmock->endpoints._buffer + dmock->endpoints._length;
+      reader1 = dds_create_reader(participant, topic1, NULL, NULL);
+      CU_ASSERT_FATAL(reader1 > 0);
+      {
+        // Check the dummy psmx instance_name.
+        dds_qos_t* qos = dds_create_qos();
+        dds_get_qos(reader1, qos);
+        uint32_t strs_len = 0;
+        char** strs = NULL;
+        CU_ASSERT_FATAL(dds_qget_psmx_instances(qos, &strs_len, &strs));
+        CU_ASSERT_FATAL(strs_len == 1 && strcmp(strs[0], "dummy_psmx") == 0);
+        free_strings(strs_len, strs);
+        dds_delete_qos(qos);
+      }
+      {
+        dds_entity* x = NULL;
+        dds_return_t rc = dds_entity_pin(reader1, &x);
+        CU_ASSERT_FATAL(rc == DDS_RETCODE_OK && dds_entity_kind(x) == DDS_KIND_READER);
+        struct dds_psmx_endpoints_set* endpt_set = &((dds_reader*)x)->m_endpoint.psmx_endpoints;
+        CU_ASSERT_FATAL(endpt_set->length == 1 && endpt_set->endpoints[0] == psmx_endpt_expected);
```

Comment:
I think the only practical way to verify this in my preferred implementation is to check in the plug-in on deleting the reader. But that needs to be done anyway.